src/Generator/TraitUsageGenerator.php

```diff
@@ -11,6 +11,12 @@
 use Reflection;
 use ReflectionMethod;
 
+/**
+ * Class TraitUsageGenerator
+ *
+ * @deprecated Deprecated in 2.7. Use UsageGenerator instead
```

Comment:
yes. we should either acknowledge that the `TraitUsageGenerator` has more responsibilities that it's name suggest (via a rename) *or* the usage functionality (for imports) should be moved to either the ClassGenerator itself or to a new class like you are suggesting.

- I think the best way forward is too create a new PR that introduces an ImportGenerator for 3.0 and simply finish this PR focussing on the has- and remove methods. This PR is getting bloated...
- do you agree that this is worth pursuing for the next release (which is soon i believe)